runtime/errors.go

```diff
@@ -74,6 +74,8 @@ func (e CallStackLimitExceededError) Error() string {
 	)
 }
 
+func (CallStackLimitExceededError) IsUserError() {}
```

Comment:
Maybe also add "assertions" the type implements the interface, here and for the other errors below:

```suggestion
var _ UserError = CallStackLimitExceededError{}

func (CallStackLimitExceededError) IsUserError() {}
```

Reply:
There are 215+ user errors, and adding the assertions manually is a lot of work. So I took the lazy approach of asserting it via a test-case 😄

Reply:
Of course! OK if I quickly add them?

Reply:
for sure, thanks!

Reply:
"quickly" 😅 added in [15943b7](https://github.com/onflow/cadence/pull/1718/commits/15943b792bd25ef03c993ad1d21488e3d77127fa)